### PR TITLE
Feature/room versioning

### PR DESCRIPTION
**Pull request by ganfra**
Implement Room versioning : #2441 
Should accept this PR first : https://github.com/matrix-org/matrix-android-sdk/pull/325

### CODE REVIEW

File: vector/src/main/java/im/vector/adapters/VectorMessagesAdapter.java [hunk 1]
```diff
@@ -796,7 +813,7 @@ public View getView(int position, View convertView, ViewGroup parent) {
         }
 
         if (null != inflatedView) {
-            inflatedView.setBackgroundColor(Color.TRANSPARENT);
+            //inflatedView.setBackgroundColor(Color.TRANSPARENT);
```
Comment (manuroe) [inline vector/src/main/java/im/vector/adapters/VectorMessagesAdapter.java]: Please remove commented code if no more useful.

File: vector/build.gradle [hunk 2]
```diff
@@ -168,6 +168,7 @@ dependencies {
     // UI
     implementation 'com.android.support.constraint:constraint-layout:1.1.0'
     implementation 'com.getbase:floatingactionbutton:1.10.1'
+    implementation 'com.binaryfork:spanny:1.0.4'
```
Comment (bmarty) [inline vector/build.gradle]: Please mention this lib in the file `open_source_licenses.html`

File: vector/src/main/java/im/vector/activity/VectorRoomActivity.java [hunk 3]
```diff
@@ -2976,24 +2996,17 @@ public void onClick(View view) {
...
-            PowerLevels powerLevels = mRoom.getState().getPowerLevels();
-
-            if (null != powerLevels) {
-                canSendMessage = powerLevels.maySendMessage(mMyUserId);
-            }
+    private boolean canSendMessages(@NonNull final RoomState state) {
+        boolean canSendMessage = !state.isVersioned();
+        if (canSendMessage) {
+            final PowerLevels powerLevels = state.getPowerLevels();
+            canSendMessage = (powerLevels != null && powerLevels.maySendMessage(mMyUserId));
```
Comment (bmarty) [inline vector/src/main/java/im/vector/activity/VectorRoomActivity.java]: With the previous algo, if powerLevels is null, user can send message. And now he cannot. Maybe it is better now, maybe it is not...

File: vector/src/main/java/im/vector/fragments/HomeFragment.java [hunk 4]
```diff
@@ -352,25 +297,4 @@ public void onLongClickRoom(View v, Room room, int position) {
...
-    }
-
-    @Override
-    public void onRoomLeft(String roomId) {
-    }
-
-    @Override
-    public void onRoomForgot(String roomId) {
-        // there is no sync event when a room is forgotten
-        initData();
```
Comment (bmarty) [inline vector/src/main/java/im/vector/fragments/HomeFragment.java]: Is it still ok without this code?

File: vector/src/main/res/layout/adapter_item_vector_message_room_versioned.xml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (bmarty) [inline vector/src/main/res/layout/adapter_item_vector_message_room_versioned.xml]: This FrameLayout is useless

File: vector/src/main/java/im/vector/fragments/HomeFragment.java (hunk 4 above)
Comment (ganfra) [reply vector/src/main/java/im/vector/fragments/HomeFragment.java]: Yes, it didn't had sense to be there

File: vector/src/main/java/im/vector/activity/VectorRoomActivity.java (hunk 3 above)
Comment (ganfra) [reply vector/src/main/java/im/vector/activity/VectorRoomActivity.java]: I think it's better :-p

File: vector/src/main/res/layout/adapter_item_vector_message_room_versioned.xml (hunk 5 above)
Comment (ganfra) [reply vector/src/main/res/layout/adapter_item_vector_message_room_versioned.xml]: Nope, it's necessary to have margin rights as ListView LayoutParams doesn't include margins :-)

**Pull request merged**